### PR TITLE
#965 use default openjfx version in GA

### LINKED ISSUE #965: GitHub Actions should use default JavaFX version
The tests that are run with GitHub Actions should by default use the static OpenJFX version that is defined internally in Substrate instead of specifying a hardcoded version.

### Expected Behavior
The tests should use the default static version of OpenJFX that is defined by Substrate as specified here: https://github.com/gluonhq/substrate/blob/master/src/main/java/com/gluonhq/substrate/Constants.java#L105

### Current Behavior
The tests are configured to use a hardcoded version of OpenJFX.

### Context
While running the tests, the native libraries are already taken from the version specified by Substrate, while the Java code is taken from the hardcoded version defined by the tests themselves. This might result in a conflict when the native methods don't match with the definitions that are defined in the Java source code. This GitHub Action run is an example of such a conflict:
https://github.com/gluonhq/substrate/runs/3045381793?check_suite_focus=true#step:12:3680

### PR DESCRIPTION
**Pull request by tiainen**
### Issue

Fixes #965

### Progress

- [ ] Change must not contain extraneous whitespace
- [ ] License header year is updated, if required
- [ ] Verify the contributor has signed [Gluon Individual Contributor License Agreement (CLA)](https://docs.google.com/forms/d/16aoFTmzs8lZTfiyrEm8YgMqMYaGQl0J8wA0VJE2LCCY)

### CODE REVIEW

File: src/test/java/com/gluonhq/substrate/IOSTest.java [hunk 1]
```diff
@@ -102,12 +104,21 @@ void testProvisioning() {
...
         boolean skipSigning = isCI();
 
+        List<String> gradleRunnerArguments = new ArrayList<>(Arrays.asList(
+            ":helloWorld:clean", ":helloWorld:build",
+            "-Dsubstrate.target=ios", "-Dskipsigning=" + skipSigning, "-DconsoleProcessLog=" + (isCI() ? "true" : "false")
+        ));
+
+        String javafxStaticSdkIos = System.getenv("JAVAFX_STATIC_SDK_IOS");
+        if (javafxStaticSdkIos != null) {
+            gradleRunnerArguments.add("-Djavafx.static.sdk=" + System.getenv("JAVAFX_STATIC_SDK_IOS"));
```
Comment (jperedadnr) [inline src/test/java/com/gluonhq/substrate/IOSTest.java:114]: Use `javafxStaticSdkIos`

File: src/test/java/com/gluonhq/substrate/IOSTest.java [hunk 2]
```diff
@@ -119,12 +130,21 @@ void helloWorldTest() {
...
         assumeTrue(!isCI());
 
+        List<String> gradleRunnerArguments = new ArrayList<>(Arrays.asList(
+            ":helloFX:clean", ":helloFX:build",
+            "-Dsubstrate.target=ios", "-DconsoleProcessLog=" + (isCI() ? "true" : "false")
+        ));
+
+        String javafxStaticSdkIos = System.getenv("JAVAFX_STATIC_SDK_IOS");
+        if (javafxStaticSdkIos != null) {
+            gradleRunnerArguments.add("-Djavafx.static.sdk=" + System.getenv("JAVAFX_STATIC_SDK_IOS"));
```
Comment (jperedadnr) [inline src/test/java/com/gluonhq/substrate/IOSTest.java:140]: Use `javafxStaticSdkIos`

**Review by johanvos — APPROVED** [review]

File: test-project/helloFX/build.gradle [hunk 3]
```diff
@@ -34,11 +34,23 @@ task runScript(type: Exec) {
...
+        imageClasspath = "${javafxStaticSdk}/lib/javafx.base.jar:" +
+                "${javafxStaticSdk}/lib/javafx.graphics.jar:" +
+                "${javafxStaticSdk}/lib/javafx.controls.jar:" +
+                project.sourceSets.main.runtimeClasspath.filter{!it.name.contains('javafx')}.asPath
+    } else {
+        imageClasspath = project.sourceSets.main.runtimeClasspath.asPath
+    }
+
+    def argsList = ["${javaHome}/bin/java", '-cp', project.configurations.substrate.asPath,
+                    "-Dverbose=true",
```
Comment (jperedadnr) [inline test-project/helloFX/build.gradle:48]: Probably only if `isCI()` is true?

File: test-project/helloFXML/build.gradle [hunk 4]
```diff
@@ -34,14 +34,25 @@ task runScript(type: Exec) {
...
+        imageClasspath = project.sourceSets.main.runtimeClasspath.asPath
+    }
+
+    def argsList = ["${javaHome}/bin/java", '-cp', project.configurations.substrate.asPath,
+                    "-Dimagecp=${imageClasspath}",
                     "-Dgraalvm=${graalvmHome}",
                     "-Dbundleslist=com.gluonhq.substrate.test.bundle",
-                    "-Dreflectionlist=javafx.fxml.FXMLLoader,com.gluonhq.substrate.test.MainController,javafx.scene.control.Button,javafx.scene.control.Label",
-            ]
+                    "-Dreflectionlist=javafx.fxml.FXMLLoader,com.gluonhq.substrate.test.MainController,javafx.scene.control.Button,javafx.scene.control.Label"]
```
Comment (jperedadnr) [inline test-project/helloFXML/build.gradle:52]: Only `com.gluonhq.substrate.test.MainController` is required here

File: test-project/helloFXML/build.gradle [hunk 5]
```diff
@@ -34,14 +34,25 @@ task runScript(type: Exec) {
...
+                "${javafxStaticSdk}/lib/javafx.graphics.jar:" +
+                "${javafxStaticSdk}/lib/javafx.controls.jar:" +
+                "${javafxStaticSdk}/lib/javafx.fxml.jar:" +
+                project.sourceSets.main.runtimeClasspath.filter{!it.name.contains('javafx')}.asPath
+    } else {
+        imageClasspath = project.sourceSets.main.runtimeClasspath.asPath
+    }
+
+    def argsList = ["${javaHome}/bin/java", '-cp', project.configurations.substrate.asPath,
+                    "-Dimagecp=${imageClasspath}",
```
Comment (jperedadnr) [inline test-project/helloFXML/build.gradle:49]: Add verbose if `isCI()`?

File: test-project/helloGluon/build.gradle [hunk 6]
```diff
@@ -48,11 +48,22 @@ task runScript(type: Exec) {
...
+        imageClasspath = "${javafxStaticSdk}/lib/javafx.base.jar:" +
+                "${javafxStaticSdk}/lib/javafx.graphics.jar:" +
+                "${javafxStaticSdk}/lib/javafx.controls.jar:" +
+                project.sourceSets.main.runtimeClasspath.filter{!it.name.contains('javafx')}.asPath
+    } else {
+        imageClasspath = project.sourceSets.main.runtimeClasspath.asPath
+    }
+
+    def argsList = ["${javaHome}/bin/java", '-cp', project.configurations.substrate.asPath,
+                    "-Dimagecp=${imageClasspath}",
```
Comment (jperedadnr) [inline test-project/helloGluon/build.gradle:62]: Verbose if `isCI()`?

**Review by jperedadnr — APPROVED** [review]

**Pull request merged**